libcontainer/cgroups/utils.go

```diff
@@ -257,7 +257,13 @@ func RemovePath(path string) error {
 	}
 
 	infos, err := os.ReadDir(path)
-	if err != nil && !os.IsNotExist(err) {
+	if err != nil {
+		if os.IsNotExist(err) {
+			// Please keep this error eraser, or else it will return ErrNotExist
+			// for cgroupv2.
+			// Please see https://github.com/opencontainers/runc/issues/4518
+			err = nil
```

Comment:
Can we `return nil` here? Resetting the `err` combined with the comment made me go look if the `err` was used in a `defer` (or if there was something special); returning `nil` here makes it more clear it's a regular "we can ignore this".

Reply:
On that matter, we should do the same below in the `infos` loop, and also return early if there's an error;


```go
for _, info := range infos {
	if info.IsDir() {
		// We should remove subcgroup first.
		if err := RemovePath(filepath.Join(path, info.Name())); err != nil {
			return err
		}
	}
}
```

That would allow use to remove the `if err == nil` check outside of the loop, and just;

```go
return rmdir(path, true)
```


❓ Question though; should the `infos` loop also ignore `os.IsNotExist()` ?


```go
	for _, info := range infos {
		if !info.IsDir() {
			continue
		}
		// We should remove subcgroup first.
		if err := RemovePath(filepath.Join(path, info.Name())); err != nil && !os.IsNotExist(err) {
			return err
		}
	}
```

Reply:
> Question though; should the `infos` loop also ignore `os.IsNotExist()` ?

I think there is no chance we will get ErrNotExist error in here.


> On that matter, we should do the same below in the `infos` loop, and also return early if there's an error;

👍